### DISCUSSION TITLE
combining data from 2 sweeps

### DISCUSSION THREAD

**Discussion by jdsorber** [ARM-DOE/pyart#1458] [Q&A] [answered]
I am trying to filter gates using both dual-pol and velocity data.  I'm running into trouble because the first sweep consists of reflectivity and dual-pol data while the second sweep consists of reflectivity and doppler velocity at the same elevation.  This creates an issue if I want to filter by both but can only plot one sweep. 

That leads to my question: is there a functionality that allows one to merge 2 sweeps together or plot by elevation angle instead of sweep number.

**Comment by jdsorber** [comment]
Adding a sweep number kwarg to the gatefilter function would also work for my purposes.

**Comment by jlpippitt** [comment, marked as answer]
Wrote this code to merge 88D split cuts.  This should help.

def merge_split_cuts(self):

    '''
    Function to merge 88D split cut sweeps and output new radar object.
    '''

    sweep_table = []
    reflectivity = []
    velocity = []

    # Create a list of elevations
    vcp = self.radar.metadata['vcp_pattern']
    print('VCP Pattern:  ', vcp)
    elist = [x.shape[0] for x in self.radar.iter_elevation()]
    n = 0
    while n <= len(elist)-1:
        
        vcp_2 = [215, 35, 212, 32, 12, 31]
        vcp_3 = [112]

        if elist[n] == 720 and elist[n+1] == 720:
            sweep_table.append((n, n+1))
            reflectivity.append(n)
            velocity.append(n+1)
            if  vcp in vcp_2:
                n += 2
            elif vcp in vcp_3:
                n += 3
        elif elist[n] == 360:
            sweep_table.append((n, n))
            reflectivity.append(n)
            velocity.append(n)
            n += 1
        elif (elist[n] == 720 or elist[n] == 360)  and (elist[n+1] != 720 or elist[n+1] != 360):
            print(elist)
            sys.exit("Bad file")

    print(" ","Merging WSR-88D Split Cuts", sep='\n')
    #print("\n Number of reflectivity levels:  %d" % len(reflectivity))
    #print("\n Number of radial velocity  levels:  %d\n" % len(velocity))
    
    #Create DZ and VR radar structures
    radar_dz = self.radar.extract_sweeps(reflectivity)
    radar_vr = self.radar.extract_sweeps(velocity)
    #print(radar_dz.fixed_angle['data'][:])
    #print(radar_vr.fixed_angle['data'][:])

    #Create new radar
    radar = copy.deepcopy(radar_dz)

    fill_value = -32767.0

    #Add VR and SW fleids to new radar
    vr_field = radar_vr.fields['VEL']['data'].copy()
    vr_dict = {'data': vr_field, 'units': '', 'long_name': 'Velocity',
               '_FillValue': fill_value, 'standard_name': 'VEL'}
    radar.add_field('VEL', vr_dict, replace_existing=True)

    sw_field = radar_vr.fields['SW']['data'].copy()
    sw_dict = {'data': sw_field, 'units': '', 'long_name': 'Spectrum Wdith',
               '_FillValue': fill_value, 'standard_name': 'SW'}
    radar.add_field('SW', sw_dict, replace_existing=True)


    print("New merged elevation angles:  ", radar.fixed_angle['data'][:], sep='\n')

    return radar

**Reply by jdsorber** [reply]
This worked, thank you! I do think that a built-in function to merge split cuts should be considered rather than having to create a new radar object.
